lib/src/native/realm_core.dart

```diff
@@ -1055,6 +1055,29 @@ class _RealmCore {
   void clearCachedApps() {
     _realmLib.realm_clear_cached_apps();
   }
+
+  List<UserHandle> getUsers(App app) {
+    return using((arena) {
+      final usersCount = arena<IntPtr>();
+      _realmLib.invokeGetBool(() => _realmLib.realm_app_get_all_users(app.handle._pointer, nullptr, 0, usersCount));
+
+      final usersPtr = arena<Pointer<realm_user>>(usersCount.value);
+      _realmLib.invokeGetBool(() => _realmLib.realm_app_get_all_users(
```

Comment:
This should handle the case where a user is added/removed between the two calls. Right now, there's a race condition.

Reply:
This is actually a bug in Core: https://github.com/realm/realm-core/issues/5430.